modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/request/body/contributor/GeneralSXPSearchRequestBodyContributor.java

```diff
@@ -40,15 +42,19 @@ public void contribute(
 			return;
 		}
 
-		if (generalConfiguration.getClauseContributorsExcludes() != null) {
+		if ((generalConfiguration.getClauseContributorsExcludes() != null) &&
+			!_hasIndexConfiguration(configuration)) {
```

Comment:
@wcao20170619 I think it would be more performant if _hasIndexConfiguration() would be called once and set to a boolean and not called 3 times

Reply:
Got that! thank you!